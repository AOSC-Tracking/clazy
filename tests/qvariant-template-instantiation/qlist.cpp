#include <QtCore/QVariant>
#include <QtCore/QUrl>


enum Foo {
    Foo1
};
Q_DECLARE_METATYPE(Foo)


void test()
{
     QList<QVariant> list;
    list.append("John Doe");
    list.append(42);
    list.value(0).toString(); //Ok
    list[1].value<int>(); //Warn
    
    list[0].value<Foo>(); //Ok
}