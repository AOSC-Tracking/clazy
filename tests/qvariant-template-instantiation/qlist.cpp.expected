qvariant-template-instantiation/qlist.cpp:17:5: warning: Use QVariant::toInt() instead of QVariant::value<int>() [-Wclazy-qvariant-template-instantiation]
